.evergreen/evergreen.yml.in

```diff
@@ -1245,46 +1817,49 @@ tasks:
   ###
   # E2E TESTS
   ###
-  <% for (const { executableOsId, compileBuildVariant } of RELEASE_PACKAGE_MATRIX) {
-    for (const mVersion of ['stable', '7.0.x', '6.0.x']) {
-      for (const fipsVariant of ['fips', 'nofips']) {
+  # Tests reuse the same compilation build variant, so we create those variations based on this.
+  <% for (const compileBuildVariant of ['build_linux_x64', 'build_linux_arm64', 'build_darwin', 'build_darwin_arm64', 'build_win32']) { 
+    const platformName = compileBuildVariant.replace('build_', '');
+    const executableOsId = platformName == 'darwin' ? 'darwin-x64' : platformName.replaceAll('_', '-');
     %>
-  - name: e2e_tests_<% out(executableOsId.replace(/-/g, '_')) %><% out(mVersion === 'stable' ? '' : '_' + mVersion.replace(/[^a-zA-Z0-9]/g, '')) %><% out(fipsVariant === 'fips' ? '_fips' : '') %>
-    tags: ["e2e-test"]
+  - name: e2e_tests_<% out(platformName) %>
+    tags: ["e2e-test", "assigned_to_jira_team_mongosh_mongosh"]
     depends_on:
       - name: compile_artifact
+      <% if (compileBuildVariant != '') { %>
         variant: <% out(compileBuildVariant) %>
+      <% } %>
     commands:
       - func: checkout
       - func: install
         vars:
-          node_js_version: "<% out(NODE_JS_VERSION_20) %>"
+          node_js_version: ${node_js_version}
       - func: download_compiled_artifact
         vars:
           executable_os_id: <% out(executableOsId) %>
       - func: run_e2e_tests
         vars:
-          node_js_version: "<% out(NODE_JS_VERSION_20) %>"
-          mongosh_server_test_version: "<% out(mVersion) %>-enterprise"
-          mongosh_test_e2e_force_fips: "<% out(fipsVariant === 'fips' ? '1' : '') %>"
-          disable_openssl_shared_config_for_bundled_openssl: ${disable_openssl_shared_config_for_bundled_openssl|false}
-  <% } } %>
-  - name: perf_tests_<% out(executableOsId.replace(/-/g, '_')) %>
+          node_js_version: ${node_js_version}
+          mongosh_server_test_version: ${mongosh_server_test_version}
+          mongosh_shared_openssl: ${mongosh_shared_openssl}
+          mongosh_test_e2e_force_fips: ${mongosh_test_e2e_force_fips}
+          disable_openssl_shared_config_for_bundled_openssl: ${disable_openssl_shared_config_for_bundled_openssl|false} 
+  <% } %>
+  - name: perf_tests
     tags: ["perf-test"]
     depends_on:
       - name: compile_artifact
-        variant: <% out(compileBuildVariant) %>
+        variant: build_linux_x64
     commands:
       - func: checkout
       - func: install
         vars:
           node_js_version: "<% out(NODE_JS_VERSION_20) %>"
       - func: download_compiled_artifact
         vars:
-          executable_os_id: <% out(executableOsId) %>
+          executable_os_id: "linux-x64"
```

Comment:
This is fine right now because we only run the one task, but I don't think it would be unreasonable for us to also add e.g. an arm64 version of this task in the future – any reason why we would want to hardcode this here instead of having the task definition available to us?